### PR TITLE
Convert scalesetvms service to SDKv2

### LINKED ISSUE #3891: Update ScaleSetVMs service to SDK v2

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Converts the scalesetvms service to azure-sdk-for-go version 2 and the `asyncpoller` framework.

**Which issue(s) this PR fixes**:

Fixes #3891

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: exp/controllers/azuremachinepoolmachine_controller.go [hunk 1]
```diff
@@ -346,9 +346,14 @@ func (ampmr *AzureMachinePoolMachineController) reconcileDelete(ctx context.Cont
 }
 
 func newAzureMachinePoolMachineReconciler(scope *scope.MachinePoolMachineScope) azure.Reconciler {
+	scaleSetVMsSvc, err := scalesetvms.NewService(scope)
+	if err != nil {
+		// TODO: handling this differently requires some refactoring
+		panic("failed to create new scalesetvms service")
```
Comment (nojnhuh) [inline exp/controllers/azuremachinepoolmachine_controller.go:352]: Is this refactoring queued up in a forthcoming PR? I'd vote to at least make sure this is ironed out before the release.

Comment (mboersma) [reply exp/controllers/azuremachinepoolmachine_controller.go:352]: No, I forgot this was in here and we should resolve it in this PR, just requires some more rewiring.

Comment (mboersma) [reply exp/controllers/azuremachinepoolmachine_controller.go:352]: /hold

Pending refactoring of the `// TODO:` comment.

Comment (mboersma) [reply exp/controllers/azuremachinepoolmachine_controller.go]: /hold cancel

That wasn't too bad, just needed return the error and update an interface and mock.

**Review by nojnhuh — COMMENTED** [review]
/lgtm

**Review by willie-yao — APPROVED** [review]
/lgtm

File: azure/services/scalesetvms/scalesetvms.go [hunk 2]
```diff
@@ -47,20 +48,24 @@ type (
...
+func NewService(scope ScaleSetVMScope) (*Service, error) {
+	client, err := newClient(scope)
+	if err != nil {
+		return nil, err
+	}
 	vmClient := virtualmachines.NewClient(scope)
 	return &Service{
-		Reconciler:   async.New(scope, client, client),
+		Reconciler: asyncpoller.New[armcompute.VirtualMachineScaleSetVMsClientUpdateResponse,
+			armcompute.VirtualMachineScaleSetVMsClientDeleteResponse](scope, nil, client),
```
[the remark this answers is not in the document]
Comment (mboersma) [reply azure/services/scalesetvms/scalesetvms.go]: I had not investigated yet--thanks for the clue!

**Review by nojnhuh — COMMENTED** [review]
/lgtm

**Pull request merged**